### DISCUSSION TITLE
Add a boolean user option for configuring whether flags in git commands are used in their long form

### DISCUSSION THREAD

**Discussion by hab25** [magit/magit#5050] [Ideas]
E.g., after enabling this toggle, when one would `magit-stage` all their unstaged files, they would then see `git add --update .` instead of `git add -u .` show up in their `magit-process` buffer.

**Comment by tarsius** [comment]
With only very few except — `-u` being one — Magit uses the long arguments. We use the long arguments for, what I assume, is the motivation behind this request; long arguments are less cryptic and better for learning and memorizing. We make a few exceptions for arguments that are used *a lot* and which Git users *should* memorize.

I won't add an option for this because then a lot of code of the form `(when (member "-x" args) ...)` would have to be replaced with (when (or (member "-x" args) (member "--long-arg" args)) ...)`.

**Reply by hab25** [reply]
> We use the long arguments for, what I assume, is the motivation behind this request; long arguments are less cryptic and better for learning and memorizing.

You assume correctly; sorry for not making it clear.

> We make a few exceptions for arguments that are used a lot and which Git users should memorize.

IMO, `-u` here shouldn't be one of those exceptions; AFAIU, the typical user very rarely has to run the command directly [^1], and AFAICT Magit does not have any transients where the `-u` flag for `git add` is displayed [^2].

And, if the above is a good idea, then it's probably also a good idea to put a task in the backlog to review all short flags usages and apply the same treatment if they fit these criteria.

> I won't add an option for this

That seems reasonable; your cost-benefit analysis is the most well-informed.
[^1]: given that Magit does a very good job of fulfilling its mission statement: "Magit aspires to be a complete Git porcelain."
[^2]: maybe `-u` can be inputted in some dynamic option, but I assume this is a very rare work workflow, and that's why Magit didn't hardcode it.
